central/config/datastore/datastore.go

```diff
@@ -225,3 +247,90 @@ func clusterRetentionConfigsEqual(c1 *storage.DecommissionedClusterRetentionConf
 	return c1.GetRetentionDurationDays() == c2.GetRetentionDurationDays() &&
 		reflect.DeepEqual(c1.GetIgnoreClusterLabels(), c2.GetIgnoreClusterLabels())
 }
+
+func (d *datastoreImpl) GetPlatformComponentConfig(ctx context.Context) (*storage.PlatformComponentConfig, bool, error) {
+	if ok, err := administrationSAC.ReadAllowed(ctx); err != nil {
+		return nil, false, err
+	} else if !ok {
+		return nil, false, nil
+	}
+
+	config, found, err := d.store.Get(ctx)
+	if config == nil {
+		return nil, false, nil
+	}
+	return config.GetPlatformComponentConfig(), found && config.GetPlatformComponentConfig() != nil, err
+}
+
+func (d *datastoreImpl) UpsertPlatformComponentConfigRule(ctx context.Context, rule *storage.PlatformComponentConfig_Rule) error {
+	if ok, err := administrationSAC.WriteAllowed(ctx); err != nil {
+		return err
+	} else if !ok {
+		return nil
+	}
+
+	config, found, err := d.store.Get(ctx)
+	if !found || err != nil {
+		return err
+	}
+	if config.PlatformComponentConfig.Rules == nil {
+		config.PlatformComponentConfig.Rules = make([]*storage.PlatformComponentConfig_Rule, 0)
+	}
+	config.PlatformComponentConfig.Rules = append(config.PlatformComponentConfig.Rules, rule)
+	return d.store.Upsert(ctx, config)
+}
+
+func (d *datastoreImpl) UpsertPlatformComponentConfigRules(ctx context.Context, rules []*storage.PlatformComponentConfig_Rule) (*storage.PlatformComponentConfig, error) {
+	if ok, err := administrationSAC.WriteAllowed(ctx); err != nil {
+		return nil, err
+	} else if !ok {
+		return nil, nil
+	}
+
+	config, found, err := d.store.Get(ctx)
+	if !found || err != nil {
+		return nil, err
+	}
+	slices.SortFunc(config.PlatformComponentConfig.Rules, ruleNameSortFunc)
+	slices.SortFunc(rules, ruleNameSortFunc)
+	if !protoutils.SlicesEqual(config.PlatformComponentConfig.Rules, rules) {
+		config.PlatformComponentConfig.NeedsReevaluation = true
+	}
+	config.PlatformComponentConfig.Rules = rules
+	err = d.store.Upsert(ctx, config)
+	if err != nil {
+		return nil, err
+	}
+	return config.PlatformComponentConfig, nil
+}
+
+func (d *datastoreImpl) DeletePlatformComponentConfigRules(ctx context.Context, rules ...string) error {
+	if ok, err := administrationSAC.WriteAllowed(ctx); err != nil {
+		return err
+	} else if !ok {
+		return nil
+	}
+
+	config, found, err := d.store.Get(ctx)
+	if !found || err != nil {
+		return err
+	}
+	if config.PlatformComponentConfig.Rules == nil {
```

Comment:
1. you should use the Getters as those are nil safe.  This could throw a nil pointer if PlatformComponentConfig is nil
2. If there are no platform config rules in the existing config, how is deleting them a valid case?  Should we not  simply return an err there?